rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,211 @@
+1. Balancing the needs of internal operations users and public-facing users.
+1. Staying as close as possible to the most basic common denominator in data across TID systems, in this case GTFS trips. This means that improvements benefit as many teams and applications as possible, and we adhere to the principle of "small pieces, loosely joined" by a common standard.
+
+The TID data architecture does not exist in a vacuum. For our purposes in this RFC, the main outside factor that we need to be aware of is the HASTUS system used for scheduling and the Plans & Schedules team that uses it to produce the new light rail schedule every rating. Challenges in this area, in particular around schedules during disruptions, will be factored in.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The below diagram gives an overall view of the proposal. Solid lines represent static data, whereas dotted lines represent realtime data:
+
+```mermaid
+  flowchart BT
+    hastus["HASTUS Schedule"]
+	gtfs["`gtfs_creator`"]
+	glides["Glides"]
+	rtr["RTR"]
+	ocs["OCS via Trike"]
+	predictions["TripUpdates"]
+	vehicles["VehiclePositions"]
+	hastus -- Trips --> gtfs
+	hastus -- Trips --> glides
+	gtfs -- Trips --> rtr
+	glides -. Trip Assignments .-> rtr
+	glides -. Trainsheet Edits .-> rtr
+	ocs -. Vehicle Locations .-> rtr
+	rtr -. Vehicle Locations .-> glides
+	rtr -. Publishes .-> predictions
+	rtr -. Publishes .-> vehicles
+```
+
+## Schedule data: HASTUS and TODS
+
+Glides will move towards using the HASTUS-provided trip ID in its schedule data. Currently, Glides [generates its own](https://github.com/mbta/glides/blob/fbe9578c8400b955304c74d34e672460a0d17a23/lib/glides/import/import_scheduled_trips.ex#L184) trip ID during the rating import process. The import code can be modified between ratings, and the next rating import will simply apply the new logic. These are the same trip IDs that GTFS ultimately uses (barring disruptions), so this means that Glides will have a foreign key it can use to communicate with other systems like RTR more easily, and the existing trip ID column can be reused. There is some subtlety about exactly what we mean by "HASTUS-provided trip ID," which is elaborated on in the [Reference-level explanation](#reference-level-explanation).
+
+In the future, Glides will move to importing scheduled trips from the [Transit Operational Data Standard (TODS)](https://ods.calitp.org/) file produced by Transit Data. This will similarly use trip IDs sourced from HASTUS and so the trip ID logic will not have to change, although the `schedule_id` we currently use will likely need to be replaced with a `service_id` to match the GTFS-based TODS standard. This change should be largely orthogonal to other changes proposed in this document, so work on TODS does not need to block adoption of these recommendations, or vice versa.
```

Comment:
In this future, would Glides get schedule data quarterly from TODS and still get personnel data from HASTUS? What if we were able to get HASTUS daily access? Would using the HASTUS-provided trip ID make it easier for us to use HASTUS daily to automatically update operators?